tool/dart_site/lib/src/commands/check_site_variable.dart

```diff
@@ -0,0 +1,52 @@
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:yaml_variable_scanner/yaml_variable_scanner.dart';
+
+import '../utils.dart';
+
+final class CheckSiteVariableCommand extends Command<int> {
+  static const String _printModeFlag = 'print-mode';
+
+  CheckSiteVariableCommand() {
+    argParser.addOption(
+      _printModeFlag,
+      help: 'Console Print Mode.',
+      allowed: ['none', 'detail', 'stats', 'detailAndStats'],
+      allowedHelp: {
+        'none': 'No content.',
+        'detail': 'Detail to file lines and columns.',
+        'stats': 'Total statistics.',
+        'detailAndStats': 'detail & stats.',
+      },
+      defaultsTo: 'detail',
+    );
+  }
+
+  @override
+  String get description =>
+      'Scan multiple files for text that can use site variables.';
+
+  @override
+  String get name => 'check-site-variable';
+
+  @override
+  Future<int> run() async {
+    final String printMode = argResults.get<String>(_printModeFlag, 'detail');

```

Comment:
Most Dart teams choose to omit local variable types. I thought we had the relevant lint enabled, but I guess not. Could you add `omit_local_variable_types` to the [`tool/dart_site/analysis_options.yaml`](https://github.com/dart-lang/site-www/blob/main/tool/dart_site/analysis_options.yaml)?

```suggestion
    final printMode = argResults.get<String>(_printModeFlag, 'detail');
```